core/src/main/scala/kafka/security/authorizer/AuthorizerWrapper.scala

```diff
@@ -175,4 +181,32 @@ class AuthorizerWrapper(private[kafka] val baseAuthorizer: kafka.security.auth.A
   override def close(): Unit = {
     baseAuthorizer.close()
   }
+
+  override def authorizeByResourceType(requestContext: AuthorizableRequestContext,
+                                       op: AclOperation,
+                                       resourceType: ResourceType): AuthorizationResult = {
+    SecurityUtils.authorizeByResourceTypeCheckArgs(op, resourceType)
+
+    if (denyAllResource(requestContext, op, resourceType)) {
+      AuthorizationResult.DENIED
+    } else if (shouldAllowEveryoneIfNoAclIsFound) {
+      AuthorizationResult.ALLOWED
+    } else {
+      super.authorizeByResourceType(requestContext, op, resourceType)
+    }
+  }
+
+  private def denyAllResource(requestContext: AuthorizableRequestContext,
+                      op: AclOperation,
+                      resourceType: ResourceType): Boolean = {
```

Comment:
The main logic of this could potentially be moved to SecurityUtils since the default Authorizer implementation, AclAuthorizer and the wrapper all do this.

Reply:
commit 8263bd3 changed the AuthorizerWrapper logic and optimized the performance a bit.

Now AuthorizerWrapper#denyAllResource will 
1. only use Authorizer#acls() to filter out the `WildcardResource` with the pattern type `LITERAL`. 
2. check if any of the filtered out bindings match the `request principle` and `request host`. 

So it's behavior diverges more from the interface default now.